app/src/main/java/org/oppia/android/app/administratorcontrols/AdministratorControlsActivityPresenter.kt

```diff
@@ -86,10 +91,47 @@ class AdministratorControlsActivityPresenter @Inject constructor(
     getAdministratorControlsFragment()!!.setSelectedFragment(lastLoadedFragment)
     activity.supportFragmentManager.beginTransaction().add(
       R.id.administrator_controls_fragment_multipane_placeholder,
-      AppVersionFragment()
+      AppVersionFragment(),
+      ADMINISTRATOR_CONTROLS_BACKSTACK
     ).commitNow()
   }
 
+  fun loadProfileEdit(profileId: Int) {
+    lastLoadedFragment = PROFILE_EDIT_FRAGMENT
+    binding.administratorControlsMultipaneOptionsBackButton!!.visibility = View.VISIBLE
+    val fragment = ProfileEditFragment.newInstance(profileId, isMultipane)
+    activity.supportFragmentManager.beginTransaction().add(
+      R.id.administrator_controls_fragment_multipane_placeholder,
+      fragment,
+      ADMINISTRATOR_CONTROLS_BACKSTACK
+    )
+      .addToBackStack(ADMINISTRATOR_CONTROLS_BACKSTACK)
+      .commit()
+  }
+
+  fun handleOnBackPressed() {
+    if (activity.supportFragmentManager.backStackEntryCount > 0)
+      activity.supportFragmentManager.popBackStackImmediate()
+  }
+
+  fun handleOnResume() {
+    activity.supportFragmentManager.addOnBackStackChangedListener {
+      if (activity.supportFragmentManager.backStackEntryCount == 0)
```

Comment:
This feels potentially fragile if we change this UI in the future.

Reply:
Umm as we have one control in the Administrator controls so I done this. What can I improve on this

Reply:
No longer relevant since you moved away from the backstack approach (though I've left a similar comment for the most recent implementation regarding asking what we should do if an unexpected fragment state is encountered).